WonderMake/NodeGraph/NodeGraph.cpp

```diff
@@ -72,36 +79,24 @@ void NodeGraph::Load()
 	PostLoad();
 }
 
-
 void NodeGraph::Compile()
 {
 	myCompiledNodeStack.clear();
 
 	CompileNodeGraph(*myRootNode, myCompiledNodeStack);
+	myNeedsRecompile = false;
 }
 
 void NodeGraph::Execute()
 {
-	//this sets up inputs/outputs
-	for (size_t i = myCompiledNodeStack.size() - 1; i < myCompiledNodeStack.size(); i--)
-	{
-		auto&& compiledNode = myCompiledNodeStack[i];
-
-		compiledNode.Node.NodeType.PrepareNode(compiledNode.Node);
-	}
-
-	for (size_t i = 0; i < myCompiledNodeStack.size(); i++)
-	{
-		auto&& compiledNode = myCompiledNodeStack[i];
-
-		compiledNode.Node.NodeType.ExecuteNodeRightToLeft(compiledNode.Node);
-	}
+	if (myNeedsRecompile)
+		Compile();
 
-	for (size_t i = 0; i < myCompiledNodeStack.size(); i++)
+	for (size_t i = myCompiledNodeStack.size() - 1; i < myCompiledNodeStack.size(); i--)
 	{
 		auto&& compiledNode = myCompiledNodeStack[i];
 
-		compiledNode.Node.NodeType.ExecuteNodeLeftToRight(compiledNode.Node);
+		compiledNode.Node.NodeType.ExecuteNode(compiledNode.Node);
 	}
```

Comment:
Why is this being done in reverse? The direction for unordered containers should not matter.